prow/git/git.go

```diff
@@ -204,7 +205,9 @@ type Repo struct {
 	git string
 	// base is the base path for remote git fetch calls.
 	base string
-	// repo is the full repo name: "org/repo".
+	// org is the organization name: "org" in "org/repo".
+	org string
```

Comment:
While tests sadly don't catch this, there are definitely some things that get broken by this change, e.G. `func (r *Repo) Push(branch string) error`

Reply:
No, that is specifically what I am changing in this PR -- it's wrong right now. See the linked issues.

Reply:
I see.
/lgtm